autogpt_platform/backend/backend/server/integrations/router.py

```diff
@@ -1,11 +1,13 @@
 import asyncio
 import logging
+from datetime import datetime, timedelta, timezone
 from typing import TYPE_CHECKING, Annotated, Awaitable, Literal
 
 from fastapi import APIRouter, Body, Depends, HTTPException, Path, Query, Request
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, SecretStr
 from starlette.status import HTTP_404_NOT_FOUND
 
+from backend.blocks.ayrshare._api import AyrshareClient, PostError, SocialPlatform
```

Comment:
Since this API stuff is used outside of `.blocks`, I'd recommend moving it to `backend/integrations/ayrshare.py`